src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -0,0 +1,279 @@
+    make_bitflags!(RLookupKeyFlag::{Override | ForceLoad | NameAlloc});
+
+/// RLookup key
+///
+/// `RLookupKey`s are used to speed up accesses in an `RLookupRow`. Instead of having to do repeated
+/// string comparisons to find the correct value by path/name, an `RLookupKey` is created using the
+/// `RLookup` which then allows `O(1)` lookup within the `RLookupRow`.
+///
+///
+/// The old C documentation for this type for posterity and later reference. Note that it is unclear
+/// how much this reflects the actual state of the code.
+///
+/// ```text
+/// RLookup Key
+///
+/// A lookup key is a structure which contains an array index at which the
+/// data may be reliably located. This avoids needless string comparisons by
+/// using quick objects rather than "dynamic" string comparison mechanisms.
+///
+/// The basic workflow is that users of a given key (i.e. "foo") are expected
+/// to first create the key by use of RLookup_GetKey(). This will provide
+/// the consumer with an opaque object that is the slot of "foo". Once the
+/// key is provided, it may then be use to both read and write the key.
+///
+/// Using a pre-defined key also allows the query to maintain a central registry
+/// of used names. If a user makes a typo in a query, this registry will easily
+/// detect that the name was not used previously.
+///
+/// Note that the same name can be registered twice, in which case it will simply
+/// increment the reference to the same key.
+///
+/// There are two arrays which are accessed to check for the key. Their use is
+/// mutually exclusive per-key, though multiple keys may exist which can access
+/// either one or the other array. The first array is the "sorting vector" for
+/// a given document. The F_SVSRC flag is set on keys which are expected to be
+/// found within the sorting vector.
+///
+/// The second array is a "dynamic" array within a given result's row data.
+/// This is used for data generated on the fly, or for data not stored within
+/// the sorting vector.
+/// ```
+#[derive(Debug, PartialEq, Eq)]
+#[pin_project]
+#[repr(C)]
+pub struct RLookupKey<'a> {
+    /// Index into the dynamic values array within the associated `RLookupRow`.
+    pub dstidx: u16,
+
+    /// If the source for this key is a sorting vector, this is the index
+    /// into the `RSSortingVector` within the associated `RLookupRow`.
+    pub svidx: u16,
+
+    /// Various flags dictating the behavior of looking up the value of this key.
+    /// Most notably, `Flags::SVSRC` means the source is an `RSSortingVector` and
+    /// `Self::svidx` should be used to look up the value.
+    pub flags: RLookupKeyFlags,
+
+    /// The path of this key.
+    ///
+    /// For fields *not* loaded from a [`FieldSpec`][ffi::FieldSpec], this points to the *same* string
+    /// as `Self::path`.
+    pub path: *const c_char,
+
+    /// The name of this key.
+    pub name: *const c_char,
+    /// The length of this key, without the null-terminator.
```

Comment:
The bytes length? (using `count_bytes()` when setting it)

Reply:
bytes yeah, updated the comment to reflect this 👍🏻